common/djangoapps/student/tests/tests.py

```diff
@@ -930,3 +932,190 @@ def test_roundtrip_with_unicode_course_id(self):
+                    {
+                        'course-v1:ManTestX+ManTest2+2014': {
+                            'run_modes_count': run_modes_count,
+                            'marketing_slug': slug,
+                            'display_name': 'Manual Test',
+                            'category': 'xseries'
+                        }
+                    },
+                    parse_data
+                )
+
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+    def test_program_courses_on_dashboard_without_configuration(self):
+        """If programs configuration is disabled than xseries upsell messages
+        will not appear on student dashboard.
+        """
+        CourseEnrollment.enroll(self.user, self.course.id)
+        self.client.login(username="jack", password="test")
+        response = self.client.get(reverse('dashboard'))
+        self.assertEquals(response.status_code, 200)
+        self.assertIn('Pursue a Certificate of Achievement to highlight', response.content)
+        self._assert_responses(response, 0)
+
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+    @ddt.data('verified', 'honor')
+    def test_modes_program_courses_on_dashboard_with_configuration(self, mode):
+        """If configuration is enabled than student can see only those course with
+        xseries upsell messages which are active in xseries programs.
+        """
+        CourseEnrollment.enroll(self.user, self.course.id, mode=mode)
+        CourseEnrollment.enroll(self.user, self.course_two.id, mode=mode)
+
+        self.client.login(username="jack", password="test")
+        self._create_program_config(enabled=True, enable_student_dashboard=True)
+
+        with patch('student.views.dummy_method_for_test_cases') as mock_data:
+            mock_data.return_value = self._create_program_data(
+                [(self.course.id, 'active'), (self.course_two.id, 'inactive')]
+            )
+            response = self.client.get(reverse('dashboard'))
+            # Count total number of courses appearing on page.
+            self.assertContains(response, 'course-container', 2)
+            self._assert_responses(response, 1)
+
+            # For verified enrollment button class will be base-btn for others it will be border-btn
+            if mode == 'verified':
+                self.assertIn('xseries-base-btn', response.content)
+            else:
+                self.assertIn('xseries-border-btn', response.content)
+
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+    @ddt.data(
+        ('inactive', 'inactive', 'inactive', 3, 0),
+        ('active', 'inactive', 'inactive', 3, 1),
+        ('active', 'active', 'inactive', 3, 2),
+        ('active', 'active', 'active', 3, 3),
+    )
+    @ddt.unpack
+    def test_different_programs_on_dashboard(self, status1, status2, status3, count, xseries_count):
```

Comment:
@zubair-arbi  I was just thinking about this.  It is difficult to avoid mentioning 'xseries' in various places due to the copy and styling that are specific to xseries, however I do think we should use `programs` instead of `xeries` in Python code / tests wherever we possibly can.  This will allow us to support other kinds of program in the future correctly and without making changes.